### PR TITLE
Add lowering for _AmdGetFuncAddr.

### PR DESCRIPTION
**Pull request by tsymalla**
This handles the 32-bit and 64-bit case.

### CODE REVIEW

**Comment by amdvlk-admin** [comment]
Can one of the admins verify this patch?

File: shared/continuations/lib/LowerRaytracingPipeline.cpp [hunk 1]
```diff
@@ -759,6 +760,32 @@ void LowerRaytracingPipelinePassImpl::handleContinuationStackIsGlobal(
...
+  (void)Consumed;
+
+  Constant *Addr = Mod->getFunction(Name);
+  if (!Addr)
+    report_fatal_error(Twine("Did not find function '") + Name +
+                       "' requested by _AmdGetFuncAddr");
+  Addr = ConstantExpr::getPtrToInt(Addr, Func.getReturnType());
+  for (auto &Use : make_early_inc_range(Func.uses())) {
+    if (auto *CInst = dyn_cast<CallInst>(Use.getUser())) {
+      if (CInst->isCallee(&Use)) {
```
Comment (jasilvanus) [inline shared/continuations/lib/LowerRaytracingPipeline.cpp:781]: I think this can use `forEachCall`.

Comment (tsymalla) [reply shared/continuations/lib/LowerRaytracingPipeline.cpp:781]: Yeah, requires to append make_early_inc_range here as well, I'll change that.

**Review by Flakebi — APPROVED** [review]

**Comment by Flakebi** [comment]
ok to test

**Review by jasilvanus — APPROVED** [review]

**Comment by amdvlk-admin** [comment]
### Test summary for commit cb8d48a8848b2861c69f261d1d51a5edf42dfbe0
<details><summary>CTS tests (Failed: 0/208496)</summary><ul><li>Built with version 1.3.5.2</li>
<details open><summary>Rhel 9.0, Gfx10</summary><ul>

<li>Passed: 36978/69503 (53.2%)</li>
<li>Failed: 0/69503 (0.0%)</li>
<li>Not Supported: 32525/69503 (46.8%)</li><li>Warnings: 0/69503 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 22.04, Navi3x</summary><ul>

<li>Passed: 41144/69503 (59.2%)</li>
<li>Failed: 0/69503 (0.0%)</li>
<li>Not Supported: 28359/69503 (40.8%)</li><li>Warnings: 0/69503 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Navi2x</summary><ul>

<li>Passed: 41178/69490 (59.3%)</li>
<li>Failed: 0/69490 (0.0%)</li>
<li>Not Supported: 28312/69490 (40.7%)</li><li>Warnings: 0/69490 (0.0%)</li>
</ul></details>


</ul></details>

**Review by Flakebi — APPROVED** [review]

**Comment by amdvlk-admin** [comment]
### Test summary for commit db1fb5f59f6417632ba56306ca58ee9caa14afd5
<details><summary>CTS tests (Failed: 0/138993)</summary><ul><li>Built with version 1.3.5.2</li>
<details open><summary>Rhel 9.0, Gfx10</summary><ul>

<li>Passed: 36978/69503 (53.2%)</li>
<li>Failed: 0/69503 (0.0%)</li>
<li>Not Supported: 32525/69503 (46.8%)</li><li>Warnings: 0/69503 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 22.04, Navi3x</summary><ul>

<li>Timeout or crash</li>

</ul></details>

<details open><summary>Ubuntu 20.04, Navi2x</summary><ul>

<li>Passed: 41178/69490 (59.3%)</li>
<li>Failed: 0/69490 (0.0%)</li>
<li>Not Supported: 28312/69490 (40.7%)</li><li>Warnings: 0/69490 (0.0%)</li>
</ul></details>


</ul></details>

**Comment by Flakebi** [comment]
retest this please
For some reason the job failed although everything passed

**Comment by amdvlk-admin** [comment]
### Test summary for commit db1fb5f59f6417632ba56306ca58ee9caa14afd5
<details><summary>CTS tests (Failed: 0/138994)</summary><ul><li>Built with version 1.3.5.2</li>
<details open><summary>Ubuntu 22.04, Navi3x</summary><ul>

<li>Passed: 41144/69504 (59.2%)</li>
<li>Failed: 0/69504 (0.0%)</li>
<li>Not Supported: 28360/69504 (40.8%)</li><li>Warnings: 0/69504 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Navi2x</summary><ul>

<li>Passed: 41178/69490 (59.3%)</li>
<li>Failed: 0/69490 (0.0%)</li>
<li>Not Supported: 28312/69490 (40.7%)</li><li>Warnings: 0/69490 (0.0%)</li>
</ul></details>


</ul></details>

**Pull request merged**